### PR TITLE
Pdcl 5919 hide features from mobile developers

### PR DESCRIPTION
**Pull request by brenthosie**
The sandbox introduced the concept of the Library Editor & Library Sandbox, but didn't consider that these features don't make sense in the context of Mobile extension development. After discussing w/ Dan Grossen, we decided it is just best to hide these features from the user.

## Description

1. Hides the menu options for Lib Editor and Lib Sandbox if your extension platform is 'mobile'.
2. Redirects any attempt to circumvent the UI to access Lib Editor or Lib Sandbox to the home page.
3. Redirects Unknown routes back to the home page
4. Introduces PLATFORMS as a constant

## How Has This Been Tested?

https://jira.corp.adobe.com/browse/PDCL-5919

### Mobile Extension Development

Screen Shot 2021-04-28 at 2 43 55 PM

### Web Extension Development

Screen Shot 2021-04-28 at 2 44 52 PM

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/app/components/Menu.jsx [hunk 1]
```diff
@@ -48,16 +49,30 @@ const Menu = ({ location }) => {
...
     }
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, []);
+  }, [location.pathname]);
+
+  const { platform } = useContext(ExtensionDescriptorContext);
+  let menuItems = [
+    { key: NAMED_ROUTES.VIEW_SANDBOX, textValue: 'View Sandbox', Icon: ViewSandboxIcon }
+  ];
+  if (platform !== PLATFORMS.MOBILE) {
```
Comment (Aaronius) [inline src/app/components/Menu.jsx:70]: Is the lib sandbox useful for `edge` extensions or should we hide it for those as well?

**Review by Aaronius — APPROVED** [review]
Nice work!

**Comment by Aaronius** [comment]
In your screenshots, the part that says "and your logic that will run within a Launch library on a Launch customer's website" is only applicable when it's a web extension. You might want to make this piece of the text conditionally shown or re-work the text somehow.

**Comment by brenthosie** [comment]
Resolves https://github.com/adobe/reactor-sandbox/issues/58

**Pull request merged**